src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -541,6 +556,37 @@ object AppDefinition {
       (cmd ^ args) || (!(cmd || args) && container)
     }
 
+  private val complyWithMigrationAPI: Validator[AppDefinition] =
+    isTrue("Scheduler AppDefinition must define required labels") { app =>
+      val isNormalApp = !app.isScheduler
```

Comment:
`val understandsMigrationProtocol = app.labels.get(AppDefinition.Labels.DcosMigrationApiVersion).exists(_.nonEmpty)`

and then:

`understandsMigrationProtocol || compliesWithMigrationApi`